packages/components/src/Checkbox/Checkbox.stories.tsx

```diff
@@ -0,0 +1,123 @@
+import type { StoryObj } from "@storybook/react";
+import React from "react";
+import Checkbox, { CheckboxInput, Label } from "./Checkbox";
+
+const defaultArgs = {
+  disabled: false,
+  label: "Checkbox",
+};
+
+export default {
+  title: "Checkbox",
+  component: Checkbox,
+  args: defaultArgs,
+  argTypes: {
+    // for some reason TS infers this as type "object",
+    // so we explicitly provide a radio control instead
+    checked: {
+      control: {
+        type: "radio",
+      },
+      options: [true, false, "indeterminate"],
+    },
+  },
+};
+
+type Args = React.ComponentProps<typeof Checkbox>;
+
+/**
+ * Controlled example to make checked stories interactive.
+ */
+function CheckedExample(args: Args) {
+  const [checked, setChecked] = React.useState(true);
+  const handleChange = () => {
+    setChecked(!checked);
+  };
+
+  return (
+    <Checkbox
+      checked={checked}
+      onChange={handleChange}
+      {...defaultArgs}
```

Comment:
Would it be better to pass `defaultArgs` as `args` on the stories using CheckedExample, instead of directly passing to the component here? That way all args are handled the same?

Reply:
makes sense 😁